app/server/middleware/makeBooking.js

```diff
@@ -27,19 +27,15 @@ const makeBooking = (req, res, next) => {
       }
     })
     .then(() => {
-      knex.select('date').from('blackout_dates').where({ listing_id: listingId }).whereIn('date', requestedDates)
+      return knex.select('date').from('blackout_dates').where({ listing_id: listingId }).whereIn('date', requestedDates)
         .then((matchedDates) => {
           if (matchedDates.length > 0) {
             throw new Error('booking conflict');
           }
         });
     })
-    .catch(() => {
-      res.status(409).end('booking conflict');
-      next();
-    })
     .then(() => {
-      knex.select('price').from('prices').where({ listing_id: listingId }).then((price) => {
+      return knex.select('price').from('prices').where({ listing_id: listingId }).then((price) => {
```

Comment:
new line before .then makes it easier to read where the promise ends and new promise begins